app/src/main-process/install-web-request-filters.ts

```diff
@@ -53,7 +55,20 @@ export function installSameOriginFilter(webRequest: WebRequest) {
 
   webRequest.onBeforeSendHeaders((details, cb) => {
     const initialOrigin = requestOrigin.get(details.id)
-    const { origin } = new URL(details.url)
+    const { origin, protocol, host } = new URL(details.url)
+
+    // If it's a WebSocket Secure request directed to a github.com subdomain,
+    // probably related to the Alive server, we need to override the `Origin`
+    // header with a valid value.
+    if (protocol === 'wss:' && /(^|\.)github\.com$/.test(host)) {
+      return cb({
+        requestHeaders: {
+          ...details.requestHeaders,
+          // TODO: discuss with Alive team a good Origin value to use here
+          Origin: 'https://desktop.github.com',
```

Comment:
As mentioned in today's meeting, this is something that still needs to be discussed to make sure we use the right `Origin` before going to beta/production, but I think this should be good to go now that the feature is enabled only for dev builds.